contracts/nois-proxy/src/contract.rs

```diff
@@ -211,10 +276,11 @@ fn execute_withdraw_all(
 fn execute_withdraw(
     deps: DepsMut,
     _env: Env,
-    _info: MessageInfo,
+    info: MessageInfo,
     amount: Coin,
 ) -> Result<Response, ContractError> {
     let config = CONFIG.load(deps.storage)?;
+    ensure_eq!(info.sender, config.manager, ContractError::Unauthorized);
```

Comment:
If we limit this to the manager, we can also pass a destination address directly, making it mire flexible